### PR TITLE
Stream metrics

### LINKED ISSUE #480: Refactor and port Flo's channel-metrics support
As a developer, I'd like to refactor and port Flo's channel-metrics support to Angular4 infrastructure.

**Acceptance:**
- Consider addressing the concerns raised in spring-cloud/spring-cloud-dataflow-ui#198
- Port the controls to Angular4 specs
- Improve test coverage

### PR DESCRIPTION
**Pull request by BoykoAlex**
Prerequisite is SCDF setup for stream metrics.
Stream definition graph view should show stream metrics data:
- Aggregate message rates for apps on links
- Instance metrics on "dots" under every stream app. Dot per instance.
Metrics data is fetched every 2 seconds. Didn't touch refresh of stream defs page...

Fixes #480

### CODE REVIEW

File: ui/src/app/streams/model/stream-metrics.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Represents a StreamMetrics.
+ *
+ * @author Alex Boyko
+ */
+
+export namespace StreamMetrics {
```
Comment (jvalkeal) [inline ui/src/app/streams/model/stream-metrics.ts]: Let's not use namespaces or interfaces here. We've used simple domain classes.

File: ui/src/app/streams/streams.service.ts [hunk 2]
```diff
@@ -152,6 +153,17 @@ export class StreamsService {
...
 
+  metrics(streamNames?: string[]): Observable<StreamMetrics.Stream[]> {
+    const options = HttpUtils.getDefaultRequestOptions();
+    if (streamNames) {
+      const params =  new URLSearchParams('', URL_QUERY_ENCODER);
+      params.append('names', streamNames.join(','));
+      options.params = params;
+    }
+    return this.http.get('/metrics/streams', options)
+      .map(res => res.json());
```
Comment (jvalkeal) [inline ui/src/app/streams/streams.service.ts]: Already commented `StreamMetrics` class but we want to return a true class here, thus you need real pojo class and not just assume json magically have same format. Also need to catch with error handler.

**Comment by BoykoAlex** [comment]
@jvalkeal Switched interfaces to POJOs, added the error handler

**Comment by BoykoAlex** [comment]
@jvalkeal @sabbyanandan I have created a new PR for this without the merge conflicts: #490
Lets look at the new PR now, I'm closing this one

**Pull request closed without merging**